confluence/client.py

```diff
@@ -485,12 +531,30 @@ def add_attachment(self, content_id, file_path, file_name=None, status=None):
         if not file_name:
             file_name = os.path.basename(file_path)
 
+        attachments = self.get_attachments(content_id, filename=file_name)
```

Comment:
I still don't think this is quite right. The update_attachment function exposed here should really take the attachment_id rather than a filename.